### DISCUSSION TITLE
rowMode array and name parameters

### DISCUSSION THREAD

**Discussion by MatthewPosgate** [vitaly-t/pg-promise#934] [Q&A]
Hi, I can't work out how to call one of the query methods (eg multi) with a combination of rowMode= array and named parameters.

**Comment by vitaly-t** [comment]
https://stackoverflow.com/questions/62060312/how-to-pass-array-as-a-parameter-for-rowmode-array-in-pg-promise

**Reply by MatthewPosgate** [reply]
Thanks for the quick reply. I'm still not much clearer. Here's an example with a db.any()

query: 'select attempts,status,locked_until from auth_ip where ip=$/ip/'
values: {ip: '123.123.123.123}
rowMode: 'array'

**Reply by vitaly-t** [reply]
`rowMode` can only be passed in as part of [ParameterizedQuery](https://vitaly-t.github.io/pg-promise/ParameterizedQuery.html) or [PreparedStatement](https://vitaly-t.github.io/pg-promise/PreparedStatement.html), both of which are formatted on the server-side, which means you have no access to `pg-promise` advanced query formatting with [Named Parameters](https://github.com/vitaly-t/pg-promise?tab=readme-ov-file#named-parameters), you only have access to the basic `$1, $2, ...` variables that PostgreSQL server understands, and that's it, that's the trade-off.

**Reply by vitaly-t** [reply]
...so you can only do:

```ts
query: 'select attempts, status, locked_until from auth_ip where ip = $1'
values: ['123.123.123.123']
rowMode: 'array'
```

**Reply by MatthewPosgate** [reply]
Thanks. Is this a limitation of the underlying pg package?

**Reply by vitaly-t** [reply]
I think No, but I'm not sure, it would take some investigation. You can always pre-format a query on the client, and then use it this way, just without values.
